modules/local/report.nf

```diff
@@ -732,6 +751,36 @@ def generate_qc_data(data, search_phrases, qc_species_tag){
 
 }
 
+
+
+def get_typing_id(sample_data, species_info, species_info_type_id, species_info_type_id_method){
+    /*
+        sample_data: The aggregated user data
+        species_info: QCParameters from the nextflow.config for the selectd species
+        species_info_type_id: The Path to the requred information in the data. Taken from the QC Parameters
+        species_info_type_id_method: The method name for the type id
+    */
+
+    def species_type_id_p = (species_info_type_id != null && species_info_type_id_method != null)
+    def species_type_prepped = (species_info_type_id != null) ^ (species_info_type_id_method != null)
+    if(!species_type_id_p){
+       return [" ", " "]
+    }
+
+    if(species_type_prepped){
+        log.warn "Both ${species_info_type_id} and ${species_info_type_id_method} must be set for a type ID to be set."
+        return [" ", " "]
+    }
```

Comment:
I'm trying to think through the `species_type_prepped` boolean calculation here.

`(species_info_type_id != null)`: if it's set (not null), then it's True.
`(species_info_type_id_method != null)`: if it's set (not null), then it's True.

`(species_info_type_id != null) ^ (species_info_type_id_method != null)`
`True OR True`
`True`

`if(species_type_prepped)`: True, both are set

`log.warn "Both ${species_info_type_id} and ${species_info_type_id_method} must be set for a type ID to be set."`

Won't this execute when both are set? Or am I misunderstanding the logic?